packages/consumption/src/modules/attributes/AttributesController.ts

```diff
@@ -599,31 +599,41 @@ export class AttributesController extends ConsumptionBaseController {
-        if (typeof predecessorSource === "undefined" || !predecessorSource.isRepositoryAttribute(this.identity.address)) {
-            return ValidationResult.error(CoreErrors.attributes.predecessorSourceAttributeIsNotRepositoryAttribute());
+        if (typeof successorSource === "undefined") {
+            return ValidationResult.error(CoreErrors.attributes.successorSourceAttributeDoesNotExist());
         }
-        if (typeof successorSource === "undefined" || !successorSource.isRepositoryAttribute(this.identity.address)) {
+
+        if (!successorSource.isRepositoryAttribute(this.identity.address)) {
             return ValidationResult.error(CoreErrors.attributes.successorSourceAttributeIsNotRepositoryAttribute());
         }
-        if (
-            typeof successorSource.succeeds === "undefined" ||
-            !predecessorSourceVersionIds.some((id) => id === successorSource.succeeds?.toString()) ||
-            typeof predecessorSource.succeededBy === "undefined" ||
-            !successorSourceVersionIds.some((id) => id === predecessorSource.succeededBy?.toString())
-        ) {
-            return ValidationResult.error(CoreErrors.attributes.successorSourceDoesNotSucceedPredecessorSource());
+
+        if (!_.isEqual(successorSource.content, successor.content)) {
+            return ValidationResult.error(CoreErrors.attributes.successorSourceContentIsNotEqualToCopyContent());
         }
 
-        const repositoryAttributeContentMatchesItsSharedAttribute =
-            _.isEqual(predecessorSource.content, predecessor.content) && _.isEqual(successorSource.content, successor.content);
-        if (!repositoryAttributeContentMatchesItsSharedAttribute) {
-            return ValidationResult.error(CoreErrors.attributes.sourceContentIsNotEqualToCopyContent());
+        let predecessorSource: any = undefined;
+        if (typeof predecessor.shareInfo.sourceAttribute !== "undefined") {
+            predecessorSource = await this.getLocalAttribute(predecessor.shareInfo.sourceAttribute);
+        }
+
+        if (typeof predecessorSource !== "undefined") {
+            if (!predecessorSource.isRepositoryAttribute(this.identity.address)) {
+                return ValidationResult.error(CoreErrors.attributes.predecessorSourceAttributeIsNotRepositoryAttribute());
+            }
+
+            const successorSourceVersionIds = (await this.getVersionsOfAttribute(successorSource.id)).map((x) => x.id.toString());
+            if (typeof predecessorSource.succeededBy === "undefined" || !successorSourceVersionIds.some((id) => id === predecessorSource.succeededBy?.toString())) {
+                return ValidationResult.error(CoreErrors.attributes.successorSourceDoesNotSucceedPredecessorSource());
+            }
+
+            if (!_.isEqual(predecessorSource.content, predecessor.content)) {
+                return ValidationResult.error(CoreErrors.attributes.predecessorSourceContentIsNotEqualToCopyContent());
+            }
```

Comment:
This error would only occur, if we ourselves had errors in the implementation of a succession or a sharing of attributes. This should be an assertion (for devs trying to catch bugs) than runtime validation of for this particular method. We're no using assertions though, so I'd say yeet it. What do you think?

Reply:
But it's also to intercept malicius requests/responses.. IMO it doesn't hurt to validate this, there can't be too much validation..